notebooks/transport/http_test.go

```diff
@@ -0,0 +1,210 @@
+package transport
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/influxdata/influxdb/v2/kit/feature"
+	"github.com/influxdata/influxdb/v2/kit/platform"
+	"github.com/influxdata/influxdb/v2/notebooks/service"
+	"github.com/influxdata/influxdb/v2/notebooks/service/mocks"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+)
+
+var (
+	orgStr       = "1234123412341234"
+	orgID, _     = platform.IDFromString(orgStr)
+	idStr        = "4321432143214321"
+	id, _        = platform.IDFromString(idStr)
+	testNotebook = &service.Notebook{
+		OrgID: *orgID,
+		ID:    *id,
+		Name:  "test notebook",
+		Spec: service.NotebookSpec{
+			"hello": "goodbye",
+		},
+	}
+	testReqBody = &service.NotebookReqBody{
+		OrgID: *orgID,
+		Name:  "Test notebook",
+		Spec: service.NotebookSpec{
+			"hello": "goodbye",
+		},
+	}
+)
+
+func TestNotebookHandler(t *testing.T) {
+	t.Parallel()
+
+	var (
+		ctrlr = gomock.NewController(t)
+		store = mocks.NewMockNotebookService(ctrlr)
+		// server needs to have a middleware to annotate the request context with the
+		// appropriate feature flags while notebooks is still behind a feature flag
+		server = annotatedTestServer(NewNotebookHandler(zap.NewNop(), store))
```

Comment:
If this is cheap to set up, we could move it into the individual sub-tests and use `zaptest.NewLogger(t)` instead of `NewNop` everywhere. Not sure if there's any logging generated that could be useful for debugging if these tests ever start to fail.

Reply:
Should be very cheap to setup (and we'll eventually get rid of it when the feature flag is not required anymore) so I made the update, along with the above change to make a new controller for every `t`.